### PR TITLE
fix: crash of file system when try to read cache dir file on android

### LINKED ISSUE #5408: getInfoAsync throws when file doesn't exist
### Environment

```
  Expo CLI 2.19.4 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 3.0.0 - /usr/local/bin/fish
    Binaries:
      Node: 10.15.3 - /var/folders/t_/7yjdv9157q79st_zkznmr7dm0000gn/T/yarn--1566419214536-0.039177485703587145/node
      Yarn: 1.17.3 - /var/folders/t_/7yjdv9157q79st_zkznmr7dm0000gn/T/yarn--1566419214536-0.039177485703587145/yarn
      npm: 6.7.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^33.0.0 => 33.0.4 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.0 => 3.11.0 
```

The issue happens in both Android and iOS.

### Steps to Reproduce

1. Call `getInfoAsync` on an inexistent file

### Expected Behavior

1. Should resolve to `{ exists: false }` instead of throwing

### Actual Behavior

1. `getInfoAsync` throws an error `File <file path> isn't readable`

### Reproducible Demo

https://snack.expo.io/@charliecruzan/getinfoasyncerror

### PR DESCRIPTION
**Pull request by numandev1**
fixed #5408
# Why
Should resolve to `{ exists: false }` instead of throwing

# How
`getInfoAsync` throws an error `File <file path isn't readable>`

# Steps to Reproduce
try to read file from `cacheDirectory` it will throw an exception which this PR has fixed

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).
### Reproducible Demo
https://snack.expo.io/@charliecruzan/getinfoasyncerror

### CODE REVIEW

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java [hunk 1]
```diff
@@ -234,10 +234,15 @@ private InputStream openResourceInputStream(String resourceName) throws IOExcept
   @ExpoMethod
   public void getInfoAsync(String uriStr, Map<String, Object> options, Promise promise) {
     try {
-      Uri uri = Uri.parse(uriStr);
-      ensurePermission(uri, Permission.READ);
+       Uri uri = Uri.parse(uriStr);
+       Uri absoluteUri = uri;
+       if ("file".equals(uri.getScheme())) {
+        uriStr = uriStr.substring(uriStr.indexOf(':') + 3);
+        absoluteUri=Uri.parse(uriStr);
```
Comment (lukmccall) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java:241]:
```suggestion
        absoluteUri = Uri.parse(uriStr);
```

**Review by lukmccall — CHANGES_REQUESTED** [review]
Looks good to me 🚀
Please add a changelog entry and add a comment in the code why we need to strip the `file` scheme.

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java (hunk 1 above)
Comment (nomi9995) [reply packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java]: I have done changes

**Comment by nomi9995** [comment]
@lukmccall i have done requested changes

File: packages/expo-file-system/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 - Enable kotlin in all modules. ([#12716](https://github.com/expo/expo/pull/12716) by [@wschurman](https://github.com/wschurman))
 
+- crash of file system when try to read cache dir file on android. ([#12716](https://github.com/expo/expo/pull/13232) by [@nomi9995](https://github.com/nomi9995))
```
Comment (lukmccall) [inline packages/expo-file-system/CHANGELOG.md:13]:
```suggestion
- Fixed crash of file system when try to read cache dir file on android. ([#12716](https://github.com/expo/expo/pull/13232) by [@nomi9995](https://github.com/nomi9995))
```

**Review by lukmccall — APPROVED** [review]

**Pull request merged**